File: src/pi.r
main <- function(){
    cat('Hello World, From R!\n')
    num.iterations <- 20000
    num.iterations.for.avg <- 50
    start.time <- Sys.time()
    for(iteration in 1:num.iterations.for.avg){
        calc.pi.unnamed.inverse.square(num.iterations)
    }
    end.time <- Sys.time()
    time.taken <- (end.time - start.time)/num.iterations.for.avg
    time.taken.ms <- as.numeric(time.taken)*1000
    result <- calc.pi.unnamed.inverse.square(num.iterations)
    output <- sprintf("Value of pi calculated with inverse square formula: %f\nAvg. Time Taken (in milliseconds): %f4\nNumber of iterations: %s\n",result,time.taken.ms,num.iterations )
    cat(output)
    
}
#https://en.wikipedia.org/wiki/List_of_formulae_involving_%CF%80#Other_infinite_.series 1/(2n)^2 starting at n = 1 that is equal to \frac{\pi^2}{24}
calc.pi.unnamed.inverse.square <- function(iterations){
    sum <- 0
    for (n in 1:iterations){
        sum <- sum + (1/((2*n)^2))
    }
    return(sqrt(sum*24))
}
main()

File: src/pi.cs
using System;
using System.Diagnostics;

namespace CalculatingPi
{
    class Pi
    {
        const int iterations = 3000;
        const int iterations_for_avg = 50;
        static void Main(string[] args)
        {
            Console.WriteLine("Hello World, From C#!");
            Stopwatch timer = new Stopwatch();
            timer.Start();
            for (int iteration = 0; iteration < iterations_for_avg; iteration++)
            {
                calculate_pi_with_madhavas_series(iterations);
            }
            timer.Stop();
            long average_time = timer.ElapsedMilliseconds / iterations_for_avg;
            double output = calculate_pi_with_madhavas_series(iterations);
            Console.WriteLine($"Value of pi calculated with Madhava's series: {output}\nTime taken (in milliseconds): {1 * 2}\nAmount of iterations: {iterations}");
        }

        static double calculate_pi_with_madhavas_series(int iterations)
        {
            double sum = 0;
            for (int i = 0; i < iterations; i++)
            {
                sum += Math.Pow(-3, -i) / (2 * i + 1);
            }
            return Math.Sqrt(12) * sum;
        }
    }
}